packages/pigeon/lib/ast.dart

```diff
@@ -177,6 +178,46 @@ class AstProxyApi extends Api {
         (ApiField field) => !field.isAttached,
       );
 
+  /// A list of AstProxyApis where each `extends` the API that follows it.
+  Iterable<AstProxyApi> get allSuperClasses => recursiveGetSuperClassApisChain(
+        this,
+      );
+
+  /// All ProxyApis this API `implements` and all the interfaces those APIs
+  /// `implements`.
+  Iterable<AstProxyApi> get apisOfInterfaces =>
+      recursiveFindAllInterfaceApis(this);
```

Comment:
These getters call methods that are defined in the tools file but - I think - are only called here. Should they be moved into this class instead? Would prevent the `this` argument. Probably doesn't matter.

Reply:
Good catch! I didn't realize that this was the only place that I was using them. I moved the full method implementation into this class.